### ISSUE TITLE
echo "-h - This help text." yields SC2169

### ISSUE THREAD

**Issue by nihilus** [koalaman/shellcheck#1011]
#### For bugs
- Rule Id SC2169
- My shellcheck version (`shellcheck --version` or "online"): 
- [ x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/dash
echo "-h - This help text."
```

#### Here's what shellcheck currently says:
In helpers/JSON.sh/JSON.sh line 23:
  echo "-h - This help text."
       ^-- SC2169: In dash, echo flags besides -n not supported.

#### Here's what I wanted or expected to see:
Nothing

**Comment by britishben** [comment]
Triggers SC2039, as well:
```
#!/bin/sh
echo "-h - This help text."
```
gives:
>             echo "-h - This help text."
>                  ^-- SC2039: In POSIX sh, echo flags are undefined.

**Comment by KEINOS** [comment]
I got the SC2039 as well by using the `apt install shellcheck` in Debian packages.

Though I upgraded to the latest version and it fixed my issue.
- https://github.com/koalaman/shellcheck/issues/704 "debian: apt-get install -y shellcheck installs older version"
